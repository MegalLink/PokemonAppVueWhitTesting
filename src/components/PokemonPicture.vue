<template>
  <div class="container-pokemon">
    <img class="hidden-pokemon" :src="imgSrc" alt="pokemon" />
    <img v-if="reveal" class="fade-in" :src="imgSrc" alt="pokemon" />
  </div>
</template>
<script>
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true,
      default: 1,
    },
    reveal: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
    },
  },
};
</script>
<style scoped>
/* Pokemon Picture */
.container-pokemon {
  height: 200px;
}

img {
  position: absolute;
  height: 200px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  transform: translateX(-50%);
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}

.hidden-pokemon {
  filter: brightness(0);
}
</style>
