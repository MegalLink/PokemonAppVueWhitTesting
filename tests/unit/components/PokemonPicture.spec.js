import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture";
describe("PokemonPicture Component Tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: { pokemonId: 1, reveal: false },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render just one image when reveal is false", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: { pokemonId: 100, reveal: false },
    });
    const [img1, img2] = wrapper.findAll("img");
    expect(img1.exists()).toBeTruthy();
    expect(img1.classes("hidden-pokemon")).toBeTruthy();
    expect(img2).toBeUndefined();
  });

  it("should render two images when reveal is true", () => {
    const pokemonID = 50;
    const wrapper = shallowMount(PokemonPicture, {
      props: { pokemonId: pokemonID, reveal: true },
    });
    const [img1, img2] = wrapper.findAll("img");
    expect(img1.exists()).toBeTruthy();
    expect(img1.classes("hidden-pokemon")).toBeTruthy();
    expect(img2.exists()).toBeTruthy();
    expect(img1.attributes("src")).toBe(
      `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonID}.svg`
    );
  });
});
