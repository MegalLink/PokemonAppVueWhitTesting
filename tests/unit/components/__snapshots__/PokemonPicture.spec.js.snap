// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Component Tests Should match snapshot 1`] = `
<div class="container-pokemon"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon">
  <!--v-if-->
</div>
`;
